hazelcast/src/main/java/com/hazelcast/instance/GroupProperty.java

```diff
@@ -548,6 +548,18 @@
      */
     QUERY_OPTIMIZER_TYPE("hazelcast.query.optimizer.type", QueryOptimizerFactory.Type.RULES.toString()),
 
+
+    /**
+     * Cache de-serialized entries. Decreases number of de-serializations needed to evaluate a query when true.
+     * It allows to trade some memory for CPU cycles.
+     *
+     * Default: true
```

Comment:
yeah, this makes sense. @ajermakovics suggested we could piggy-back on the `optimizeQueries` option - the same deserialized object could be shared by both `CachedDataRecord` and `CacheQueryEntry`. I'm not sure if that's doable, but I'll have a look.